src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDiffieHellmanCng.cs

```diff
@@ -63,6 +63,45 @@ private byte[] ExportFullKeyBlob(bool includePrivateParameters)
                     return CngKeyLite.ExportKeyBlob(keyHandle, blobType);
                 }
             }
+
+            private byte[] ExportEncryptedPkcs8(ReadOnlySpan<char> pkcs8Password, int kdfCount)
+            {
+                using (SafeNCryptKeyHandle keyHandle = GetDuplicatedKeyHandle())
+                {
+                    return CngKeyLite.ExportPkcs8KeyBlob(keyHandle, pkcs8Password, kdfCount);
+                }
+            }
+
+            private bool TryExportEncryptedPkcs8(
+                ReadOnlySpan<char> pkcs8Password,
+                int kdfCount,
+                Span<byte> destination,
+                out int bytesWritten)
+            {
+                using (SafeNCryptKeyHandle keyHandle = GetDuplicatedKeyHandle())
+                {
+                    return CngKeyLite.TryExportPkcs8KeyBlob(
+                        keyHandle,
+                        pkcs8Password,
+                        kdfCount,
+                        destination,
+                        out bytesWritten);
+                }
+            }
+
+            private void AcceptImport(CngPkcs8.Pkcs8Response response)
+            {
+                SafeNCryptKeyHandle keyHandle = response.KeyHandle;
+
+                _key.SetHandle(
+                    keyHandle,
+                    CngKeyLite.GetPropertyAsString(
```

Comment:
Can this throw?  Just wondering if keyHandle would need to be disposed aggressively in that case.

Reply:
It can't throw under any reasonable situations.  If we OOM on allocating the string I'm fine with letting keyHandle get finalized.